web/src/main/java/org/springframework/security/web/DefaultRedirectStrategy.java

```diff
@@ -56,7 +62,13 @@ public void sendRedirect(HttpServletRequest request, HttpServletResponse respons
 		if (this.logger.isDebugEnabled()) {
 			this.logger.debug(LogMessage.format("Redirecting to %s", redirectUrl));
 		}
-		response.sendRedirect(redirectUrl);
+		if (this.statusCode == null) {
+			response.sendRedirect(redirectUrl);
+		}
+		else {
+			response.setHeader(HttpHeaders.LOCATION, redirectUrl);
+			response.setStatus(this.statusCode.value());
```

Comment:
To ensure that the commit expectations are similar, please add a call to `response.getWriter().flush()`.